leases/models.py

```diff
@@ -115,6 +115,13 @@ class Meta:
         verbose_name_plural = _("berth leases")
         default_related_name = "berth_leases"
 
+    def save(self, *args, **kwargs):
+        if self.start_date.year != self.end_date.year:
```

Comment:
we should maybe do it in the `clean()` method? and call `self.full_clean()` from `save()`. like we do [here](https://github.com/City-of-Helsinki/berth-reservations/blob/develop/applications/models.py#L256)